src/Stratis.Bitcoin.Features.SmartContracts/Consensus/SmartContractPosTransaction.cs

```diff
@@ -0,0 +1,24 @@
+﻿using NBitcoin;
+using NBitcoin.DataEncoders;
+using NBitcoin.Protocol;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.Consensus
+{
+    /// <summary>A smart contract proof of stake transaction.</summary>
+    public class SmartContractPosTransaction : PosTransaction
```

Comment:
Do we need these classes? I get the PosTransaction may have different params in future, but SC transactions don't?

Reply:
Agree, if we're not changing anything in an inherited class I don't think it's needed.

Reply:
There are some of the POS consensus rules which checks the transaction type, where if its not SC POS tx we execute it, which makes the rule fail. Not the best design I know but for now we have to do it like this ;)